crates/circuit/src/circuit_instruction.rs

```diff
@@ -762,6 +766,45 @@ impl<'py> FromPyObject<'py> for OperationFromPython {
             });
         }
 
+        // We need to check by name here to avoid a circular import during initial loading
+        if ob.getattr(intern!(py, "name"))?.extract::<String>()? == "unitary" {
+            let params = extract_params()?;
+            if let Param::Obj(data) = &params[0] {
+                let py_matrix: PyReadonlyArray2<Complex64> = data.extract(py)?;
+                let matrix: Option<MatrixView2<Complex64>> = py_matrix.try_as_matrix();
+                if let Some(x) = matrix {
+                    let unitary_gate = UnitaryGate {
```

Comment:
Is it better to make this variable be the `Box<UnitaryGate>`? I imagine the compiler will figure out that you want this to end up on the heap and thus avoid a copy from stack to heap, but maybe having it locally opens us up to someone in the future using / modifying it on the stack, preventing that optimization.

Reply:
At least in this context since we immediately pack it I wasn't thinking there was anything we could do locally in this context. I was thinking the compiler would see that nothing is touching the matrix after it's created. But I can immediately box it here if you think that's better.

Reply:
I'm not especially worried if you leave it as is, though I believe we create the box immediately for the other pointer codepaths.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13759/commits/ffb2ecf703ac776ffcc0fc6663b34dee81242530